main.py

```diff
@@ -1,19 +1,20 @@
 """Proxy server to communicate a client to ARTIQ."""
 
 import ast
+import asyncio
+import copy
+import functools
 import importlib.util
 import json
 import logging
 import os
 import posixpath
 import shutil
 import time
-import copy
-import asyncio
 from contextlib import asynccontextmanager
 from datetime import datetime
 from enum import Enum
-from typing import Any, Dict, List, Optional, Union
+from typing import Any, Dict, List, Tuple, Optional, Union
```

Comment:
It is too minor, but could you exchange "Tuple" and "Optional" to sort correctly?